Nk/feature user data model

NatalyMac opened the pull request:
add fetching user data /user, /users, /users/id
{
  "data": {
    "first_name": "Paula",
    "last_name": "Schmidt",
    "email": "beverly.grimes@example.org",
    "city": "",
    "country": "",
    "birthday": "",
    "local_id": "",
    "avatar": {
      "urlAva": "",
      "thumbnailUrlAva": ""
    },
    "id": 331,
    "global_id": "567920fc215c82267c1a3cc3",
    "display_name": "jason63",
    "reputation": 212,
    "status_id": 19,
    "last_visit_at": "2015-10-01 07:33:29",
    "role_id": 19,
    "role": "User",
    "status": "online"
  },
  "_meta": []
}

change curl with get_file_content because of the trouble  with double  sending curl  - authentication fail from auth server

NatalyMac committed:
fetch user, users data

NatalyMac committed:
add globalIds in user seed

NatalyMac committed:
Merge branch 'next' of https://github.com/BinaryStudioAcademy/bsa-2016-forum into NK/feature-UserDataModel

NatalyMac committed:
Merge branch 'next' of https://github.com/BinaryStudioAcademy/bsa-2016-forum into NK/feature-UserDataModel

NatalyMac committed:
Merge branch 'next' of https://github.com/BinaryStudioAcademy/bsa-2016-forum into NK/feature-UserDataModel

NatalyMac committed:
Merge branch 'next' of https://github.com/BinaryStudioAcademy/bsa-2016-forum into NK/feature-UserDataModel

NatalyMac committed:
change curl with get_file_content

NatalyMac committed:
Merge branch 'next' of https://github.com/BinaryStudioAcademy/bsa-2016-forum into NK/feature-UserDataModel

binaryvo commented on app/Providers/AppServiceProvider.php:
```diff
@@ -46,6 +46,8 @@ public function boot()
-        //
+        $this->app->bind(
+            'Repositories\Contracts\UserStoreInterface',
+            'Repositories\UserStore');
```
Move closing parenthesis to next string please (psr-2)

binaryvo commented:
Why have you changed permission modes for next files:
bootstrap/app.php
bootstrap/autoload.php
bootstrap/cache/.gitignore
?

NatalyMac commented:
Why have you changed permission modes for next files
- sometimes I have Laravel fails with files permission - storage and bootstrap. And I've changed them. I will change them back

I've changed them to 644 and now I have 
warning: «bootstrap/.gitignore» недоступно: Отказано в доступе
bootstrap/app.php: Отказано в доступе
bootstrap/autoload.php: Отказано в доступе
bootstrap/cache/.gitignore: Отказано в доступе

when I try git add -A
what should I do?

NatalyMac committed:
fix )

NatalyMac committed:
Merge branch 'next' of https://github.com/BinaryStudioAcademy/bsa-2016-forum into NK/feature-UserDataModel

binaryvo commented on app/Repositories/UserStore.php:
```diff
@@ -0,0 +1,126 @@
+        $userProfile['role_id'] = $user->role_id;
+
+        $role = \DB::table('roles')->where('id', $user->role_id)->value('name');
+        $status = \DB::table('user_statuses')->where('id', $user->status_id)->value('name');
```
Please use models instead

binaryvo commented on app/Repositories/UserStore.php:
```diff
@@ -0,0 +1,126 @@
+                    $r['last_visit_at'] = $userInner['last_visit_at'];
+                    $r['role_id'] = $userInner['role_id'];
+
+                    $role = \DB::table('roles')->where('id', $r['role_id'])->value('name');
```
Please use models instead

binaryvo commented on app/Repositories/UserStore.php:
```diff
@@ -0,0 +1,126 @@
+                    $r['local_id'] = $userFunc->id;
+                    $r['avatar'] = $userFunc->avatar;
+                    return $r;
+                },
```
I guess it could be possible to move mapper to separate function to prevent repetitive code

binaryvo commented:
Please fix all remarks and make PR again
